bundles/org.platformio.eclipse.ide.home/src/org/platformio/eclipse/ide/home/api/Environment.java

```diff
@@ -20,13 +20,16 @@
  *******************************************************************************/
 package org.platformio.eclipse.ide.home.api;
 
+import java.io.File;
 import java.nio.file.Path;
 import java.util.List;
 
 public interface Environment {
 
 	CommandResult execute(String command, List<String> arguments);
 
+	CommandResult execute(String command, List<String> arguments, File workingDirectory);

```

Comment:
Interface is getting polluted
It looks like you need a class to aggregate all this parameters